lib/sw_indexer_engine.rb

```diff
@@ -32,4 +35,9 @@ def index(druid, targets = nil)
   def purl_model(druid)
     DiscoveryIndexer::InputXml::Purlxml.new(druid).load
   end
+
+  def update_marc_record(druid)
```

Comment:
I wonder if this logic ought to move into the model itself? It might be easier to test there, if nothing else.